metacat-common-server/src/main/java/com/netflix/metacat/common/server/connectors/SpringConnectorFactory.java

```diff
@@ -30,20 +30,24 @@
  */
 
 public abstract class SpringConnectorFactory implements ConnectorFactory {
-    protected final String catalogName;
     protected final AnnotationConfigApplicationContext ctx;
+    private final String catalogName;
+    private final String catalogShardName;
 
     /**
      * Constructor.
      *
      * @param catalogName            catalog name
+     * @param catalogShardName       catalog shard name
      * @param connectorInfoConverter connector info converter
      * @param connectorContext       connector related config
      */
     public SpringConnectorFactory(final String catalogName,
+                                  final String catalogShardName,
```

Comment:
If this can be null I'd recommend flagging it with `@Nullable`

Reply:
No, it is never null.